packages/edit-post/src/components/sidebar/plugin-pre-publish-panel/test/index.js

```diff
@@ -11,21 +15,19 @@ import PluginPrePublishPanel from '../';
 
 describe( 'PluginPrePublishPanel', () => {
 	test( 'renders fill properly', () => {
-		const div = document.createElement( 'div' );
 		render(
 			<SlotFillProvider>
 				<PluginPrePublishPanel
 					className="my-plugin-pre-publish-panel"
 					title="My panel title"
-					initialOpen={ true }
+					initialOpen
 				>
 					My panel content
 				</PluginPrePublishPanel>
 				<PluginPrePublishPanel.Slot />
-			</SlotFillProvider>,
-			div
+			</SlotFillProvider>
 		);
 
-		expect( div.innerHTML ).toMatchSnapshot();
+		expect( screen.getByText( 'My panel title' ) ).toBeInTheDocument();
```

Comment:
I've found that `.toBeVisible()` is even a bit more specific than `.toBeInTheDocument()`, but that's just me nitpicking, feel free to ignore it.

Reply:
Good idea, fixed.

By the way here's one place where our tests are happy with an invisible element, although they shouldn't be:

https://github.com/WordPress/gutenberg/blob/bf6eddd8b6342e5f7f2e68e0709600555a0f18fd/packages/block-editor/src/components/url-popover/test/__snapshots__/index.js.snap#L8

That `components-popover` element is made visible with an animation, and this snapshot caught only the initial frame of the animation, where there is `opacity: 0`. It should wait for the animation to finish. Or specify `animate={ false }`. In that test the animation wouldn't even run because there are fake timers and nobody advances them.